server/polar/subscription/service.py

```diff
@@ -1433,5 +1433,28 @@ async def mark_past_due(
 
         return subscription
 
+    async def mark_active(
+        self, session: AsyncSession, subscription: Subscription
+    ) -> Subscription:
+        """Mark a subscription as active. Used when payment succeeds after being past due."""
+
+        previous_status = subscription.status
+        previous_ends_at = subscription.ends_at
+
+        repository = SubscriptionRepository.from_session(session)
+        subscription = await repository.update(
+            subscription, update_dict={"status": SubscriptionStatus.active}
+        )
+
+        await self._after_subscription_updated(
```

Comment:
I think we should take a look at the communications (in another task). Once a customer payment fails:

- Stripe sends the communication that the payment failed
- We send a communication that the subscription is cancelled


When a payment is recovered:
- We send a communication that the subscription was created

Reply:
Yes, lot of emails to implement 😁